aten/src/ATen/native/cuda/TensorShape.cu

```diff
@@ -0,0 +1,399 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Dispatch.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/ops/empty.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <c10/util/Logging.h>
+
+namespace at::native {
+
+namespace detail {
+
+// NOTE [CUDA fast path for split_with_sizes_copy.out]
+// split_with_sizes_copy.out for contiguous operands has the following
+// properties:
+// - Each src split consists of multiple chunks that are separated by a fixed
+// stride. The number of chunks and the strides are the same across all src
+// splits.
+// - Each dst split is the concatenation of the chunks in its corresponding src
+// splits.
+// - The sizes chunks vary across splits.
+// - The alignment varies across chunks. A (src, dst) chunk pair is not
```

Comment:
To check, are the two sentences here separate?
- "The alignment varies across chunks." -> Different chunks have different alignments.
- "A (src, dst) chunk pair is not guaranteed to have the same alignment." -> Within one chunk, the src and dst may not have the same alignment.

Reply:
Yeah it's a bit confusing. What matters is the second part. Let me remove the first part.